packages/types/src/modules-sdk/module-provider.ts

```diff
@@ -0,0 +1,11 @@
+import { Constructor } from "./index"
+
+export type ModuleProviderExports = {
+  services: Constructor<any>[]
+}
+
+export type ModuleProvider = {
+  resolve: string | ModuleProviderExports
+  provider_name?: string
+  options: Record<string, unknown>
```

Comment:
todo: probably some doc would be good to explain to the user what impact it has and what kind of options are expected